### PR TITLE
Add function to copy struct of amconfig.Config

### PR DESCRIPTION
**Pull request by lelenanam**
Marshal/unmarshal json to deep copy config

### CODE REVIEW

**Comment by lelenanam** [comment]
Fixes https://github.com/weaveworks/notification/issues/109

**Comment by jpellizzari** [comment]
I created this PR, which does the same thing: https://github.com/weaveworks/cortex/pull/590

@lelenanam 's PR looks nicer. @bboreham Any preference?

**Comment by lelenanam** [comment]
My opinion is this solution is more robust and will work even if we add/change other fields of config or if fields are changing somewhere else in the code but we just haven't noticed it yet.

File: pkg/alertmanager/multitenant.go [hunk 1]
```diff
@@ -401,7 +402,25 @@ func (am *MultitenantAlertmanager) transformConfig(userID string, amConfig *amco
 		}
 	}
 
-	return &newConfig, nil
+	return newConfig, nil
+}
+
+func copyConfig(config *amconfig.Config) (*amconfig.Config, error) {
```
Comment (bboreham) [inline pkg/alertmanager/multitenant.go]: maybe add a comment explaining we need to do a deep copy because of pointers, slices, etc

**Pull request merged**